### PR TITLE
Latest ci errors

### PR DESCRIPTION
**Pull request by Vaishnavi-infoblox**
Fixing sanity errors which are related to doc_fragments and argument spec errors.

### CODE REVIEW

File: tests/sanity/ignore.txt [hunk 1]
```diff
@@ -550,3 +550,9 @@ test/units/utils/test_shlex.py future-import-boilerplate
 test/units/utils/test_shlex.py metaclass-boilerplate
```
Comment (anshulbehl) [inline tests/sanity/ignore.txt:550]: Please rename the file as ignore-2.10.txt for ansible-test to ignore these lines with ansible-2.10
Similar file should be maitained for ansible-2.9 as well as certification test will happen on ansible-2.9, but as Felix said you should also avoid ignore file and try to fix the error like I did in the last pull request unless absolutely necessary

File: tests/sanity/ignore.txt [hunk 2]
```diff
@@ -550,3 +550,9 @@ test/units/utils/test_shlex.py future-import-boilerplate
...
 test/utils/shippable/check_matrix.py replace-urlopen
 test/utils/shippable/timing.py shebang
+plugins/modules/nios_a_record.py:0:0: doc-required-mismatch: Argument 'host' in argument_spec found in provider is not required, but is documented as being required
```
Comment (anshulbehl) [inline tests/sanity/ignore.txt:553]:
```
plugins/modules/nios_a_record.py validate-modules:doc-required-mismatch
```
It should look like above for the selected test, can you make the changes and do a re-run

File: tests/sanity/ignore.txt (hunk 1 above)
Comment (Vaishnavi-infoblox) [reply tests/sanity/ignore.txt:550]: I have made the above modifications - renaming and added ignore statements but still I see 306 errors have not reduced.

File: tests/sanity/ignore.txt (hunk 2 above)
Comment (Vaishnavi-infoblox) [reply tests/sanity/ignore.txt]: Done but it didn't reduce the no of sanity errors.

Comment (anshulbehl) [reply tests/sanity/ignore.txt]: ahh, just saw this, let me check this on my end and confirm

File: tests/sanity/ignore.txt [hunk 3]
```diff
@@ -550,9 +550,11 @@ test/units/utils/test_shlex.py future-import-boilerplate
 test/units/utils/test_shlex.py metaclass-boilerplate
 test/utils/shippable/check_matrix.py replace-urlopen
 test/utils/shippable/timing.py shebang
-plugins/modules/nios_a_record.py:0:0: doc-required-mismatch: Argument 'host' in argument_spec found in provider is not required, but is documented as being required
-plugins/modules/nios_a_record.py:0:0: doc-required-mismatch: Argument 'ipv4addr' in argument_spec is not required, but is documented as being required
-plugins/modules/nios_a_record.py:0:0: doc-required-mismatch: Argument 'view' in argument_spec is not required, but is documented as being required
-plugins/modules/nios_a_record.py:0:0: invalid-ansiblemodule-schema: AnsibleModule.argument_spec.ipv4addr.ib_req: extra keys not allowed @ data['argument_spec']['ipv4addr']['ib_req']. Got True
-plugins/modules/nios_a_record.py:0:0: invalid-ansiblemodule-schema: AnsibleModule.argument_spec.name.ib_req: extra keys not allowed @ data['argument_spec']['name']['ib_req']. Got True
+plugins/modules/nios_a_record.py:0:0: doc-required-mismatch
```
Comment (anshulbehl) [inline tests/sanity/ignore.txt:553]: @Vaishnavi-infoblox These lines should be part of ignore-2.10.txt file and not ignore.txt file

**Comment by anshulbehl** [comment]
Also @Vaishnavi-infoblox @saileshgiri , Is it possible for you to run locally against ansible-2.9 as that is what we check for certification as of today
If not locally, I think you can also setup a new run with ansible-2.9, the current one is setup against ansible-2.10

The pull request I send fixes a lot of issues against ansible-2.9 and the same pull request can be reference to fix pending issues with ansible-2.9 without adding any ignore statements
Let me know if this helps, or we can get on a call as well

**Comment by Vaishnavi-infoblox** [comment]
I have reverted the changes for ignore.txt file and ignore-2.10.txt already has these changes, After all these changes, there is no reduction in the number of sanity errors. Also my VM has ansible 2.9.11. I will also go through your Pull request and try to follow the same.

File: tests/sanity/ignore.txt (hunk 3 above)
Comment (anshulbehl) [reply tests/sanity/ignore.txt]: @Vaishnavi-infoblox The location of these files needs to be inside the collection structure which in your case should be `ansible_collections/infoblox/nios_modules/tests/sanity/ignore-2.10.txt` or `ansible_collections/infoblox/nios_modules/tests/sanity/ignore-2.9.txt`

I also see a lot of unrequired lines in your sanity files please remove all of those and start with a basic lines as below:
```
plugins/modules/nios_a_record.py validate-modules:doc-required-mismatch
plugins/modules/nios_aaaa_record.py  validate-modules:doc-required-mismatch
plugins/modules/nios_a_record.py  validate-modules:module-invalid-version-added
plugins/modules/nios_a_record.py  validate-modules:invalid-ansiblemodule-schema
plugins/modules/nios_aaaa_record.py  validate-modules:doc-required-mismatch
plugins/modules/nios_aaaa_record.py  validate-modules:invalid-ansiblemodule-schema
plugins/modules/nios_aaaa_record.py  validate-modules:module-invalid-version-added
plugins/modules/nios_zone.py validate-modules:doc-required-mismatch
plugins/modules/nios_fixed_address.py validate-modules:doc-missing-type
plugins/modules/nios_fixed_address.py validate-modules:invalid-ansiblemodule-schema
```
for the 2.10 ignore file. There is a slight change in schema that I missed last time
please check if this helps

**Comment by Vaishnavi-infoblox** [comment]
Hi @anshulbehl,

Now all sanity errors are resolved please have a look and let us know if we can merge to main branch.

**Comment by anshulbehl** [comment]
@Vaishnavi-infoblox looks like everything is passing now, you can now merge into the main branch and create a collection that can be uploaded to Galaxy

**Pull request merged**